### PR TITLE
0.6 release

### LINKED ISSUE #1940: Release 0.6
It's been a while, we should definitely do a new release!

**Please comment if you have something you would like to get in this release.**

Things to merge/solve before release:

- [x] #1928 
- [x] #1945 
- [x] #1888 
- [x] #1883 
- [x] #1622 ?
- [x] #1975
- [x] #1979 
- [x] #2009
- [x] #2027 

Release tasks:
* [x] #1970 #2039
* [x] bump version
* [x] make sure installation & contributing instructions are up to date
* [x] make sure community information is up to date (e.g. in light of #2009)
* [x] Play through tutorials to make sure we haven't accidentally introduced any regressions that new players might run into
* [ ] look into building OSX or Windows binaries?
* [x] Quickly skim through commits since 0.5 release (Nov 2023) to see if there are any that ought to be labeled `CHANGELOG` but are not 
* [x] add entry to CHANGELOG
  * query to get CHANGELOG PRs:
    https://github.com/swarm-game/swarm/pulls?q=is%3Apr+closed%3A%3E2023-11-01+label%3ACHANGELOG
  * On cmdline: `gh pr list -s closed -S 'closed:>2023-11-01 label:CHANGELOG' | tr '\t' ',' > 0.6-changes.txt`
* [x] regenerate wiki cheatsheets
* [ ] tag Swarm version
* [ ] tag VSCode plugin
* [ ] write blog post on Swarm website
* [ ] publicize blog post

---

Release PRs:
* #1980 
* https://github.com/swarm-game/swarm-game.github.io/pull/6

---

Previous release Issues:
* #1543 
* #1316
* #1012
* #666

### PR DESCRIPTION
**Pull request by byorgey**
Closes #1940.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,88 @@
...
+
+* `volume` command measuring the enclosed area around a given location ([#1747](https://github.com/swarm-game/swarm/pull/1747))
+* Type synonyms ([#1865](https://github.com/swarm-game/swarm/pull/1865))
+* Recursive types ([#1894](https://github.com/swarm-game/swarm/pull/1894))
+* Entity tags and related commands ([#1635](https://github.com/swarm-game/swarm/pull/1635))
+* `meetAll` command now returns a list ([#1999](https://github.com/swarm-game/swarm/pull/1999))
+
+#### Entities + recipes
+
+* New `lens` entity and recipe for `detonator` ([#1876](https://github.com/swarm-game/swarm/pull/1876))
```
Comment (xsebek) [inline CHANGELOG.md:41]: The `sow` command from #1817 should be listed here as well, right?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,88 @@
...
+* Popups for new achievements, recipes, and commands
+  ([#2027](https://github.com/swarm-game/swarm/pull/2027))
+* Save current REPL input with down arrow
+  ([#2000](https://github.com/swarm-game/swarm/pull/2000))
+* Customizable keybindings ([#1979](https://github.com/swarm-game/swarm/pull/1979))
+
+#### Command line options
+
+* `format` subcommand now tries to preserve comments ([#1845](https://github.com/swarm-game/swarm/pull/1845))
+* Add `format --v0.5` option to port code from older syntax ([#1851](https://github.com/swarm-game/swarm/pull/1851))
```
Comment (xsebek) [inline CHANGELOG.md:79]: Please add command line `keybindings` and `docs recipes --forward`.

**Review by xsebek — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,99 @@
...
+* Recipes for `rolex`, `olfactometer`, `dozer blade`, and `hourglass` entities
+  ([#2028](https://github.com/swarm-game/swarm/pull/2028), [#2025](https://github.com/swarm-game/swarm/pull/2025))
+
+#### New scenarios
+
+* Beekeeping scenario ([#1599](https://github.com/swarm-game/swarm/pull/1599))
+* Fishing scenario ([#1628](https://github.com/swarm-game/swarm/pull/1628))
+* Dim sum restaurant ([#1686](https://github.com/swarm-game/swarm/pull/1686))
+* Gallery scenario ([#1760](https://github.com/swarm-game/swarm/pull/1760))
+
```
Comment (kostmo) [inline CHANGELOG.md:60]:
```suggestion
* Snake automata ([#1699](https://github.com/swarm-game/swarm/pull/1699))

```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,99 @@
...
+#### World features
+
+* Path caching, for more efficient repeated use of the `path` command ([#1595](https://github.com/swarm-game/swarm/pull/1595))
+* Code to render any map to PNG ([#1632](https://github.com/swarm-game/swarm/pull/1632))
+* Extensible terrain ([#1775](https://github.com/swarm-game/swarm/pull/1775))
+* Spreadable plant growth ([#1817](https://github.com/swarm-game/swarm/pull/1817))
```
Comment (kostmo) [inline CHANGELOG.md:66]:
```suggestion
#### Scenario mechanics and authoring improvements

* Path caching, for more efficient repeated use of the `path` command ([#1595](https://github.com/swarm-game/swarm/pull/1595))
* Render any map to PNG ([#1632](https://github.com/swarm-game/swarm/pull/1632))
* Extensible terrain ([#1775](https://github.com/swarm-game/swarm/pull/1775))
* Spreadable plant growth ([#1817](https://github.com/swarm-game/swarm/pull/1817))
* Exercising commands can consume entities ([#1777](https://github.com/swarm-game/swarm/pull/1777))
* Recognize built structures ([#1579](https://github.com/swarm-game/swarm/pull/1579))
* Test for point-to-point connectivity ([#1721](https://github.com/swarm-game/swarm/pull/1721))
* Validate palettes ([#1938](https://github.com/swarm-game/swarm/pull/1938)), map shape ([#1935](https://github.com/swarm-game/swarm/pull/1935)), subworld references ([#1875](https://github.com/swarm-game/swarm/pull/1875))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,99 @@
...
+Some of the highlights of this release include native Windows support,
+customizable keybindings, type synonyms and equirecursive types, and a
+prototype tournament server
+([#1798](https://github.com/swarm-game/swarm/pull/1798)) where players
+can upload their solutions to challenge scenarios.  See below for a
+more detailed list (or see the [complete list of git
+commits](https://github.com/swarm-game/swarm/commits/main/?since=2023-11-01)).
+
+Aside from the external changes listed below (including a few major
+additions), this release cycle saw a *lot* of internal refactoring.
```
Comment (kostmo) [inline CHANGELOG.md:14]:
```suggestion
Aside from the more visible changes listed below, this release cycle saw a *lot* of internal refactoring.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,99 @@
...
+For example, we finished splitting the codebase into a number of
+independent sublibraries, split several large modules into smaller
+modules, and completed a number of additional simplifications and
+improvements.
```
Comment (kostmo) [inline CHANGELOG.md:18]: Either be more specific on what kind of improvements, or remove.  E.g.
```suggestion
code quality improvements.
```

**Review by kostmo — APPROVED** [review]

**Pull request merged**